injection/sharedmain/main.go

```diff
@@ -143,102 +166,180 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
-	} else if !apierrors.IsNotFound(err) {
-		logger.With(zap.Error(err)).Fatalf("Error reading ConfigMap %q", logging.ConfigMapName())
-	}
+		// Watch the logging config map and dynamically update logging levels.
+		if _, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(logging.ConfigMapName(),
+			metav1.GetOptions{}); err == nil {
+			cmw.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
+		} else if !apierrors.IsNotFound(err) {
+			logger.With(zap.Error(err)).Fatalf("Error reading ConfigMap %q", logging.ConfigMapName())
+		}
 
-	// Watch the observability config map
-	if _, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(metrics.ConfigMapName(),
-		metav1.GetOptions{}); err == nil {
-		cmw.Watch(metrics.ConfigMapName(),
-			metrics.UpdateExporterFromConfigMap(component, logger),
-			profilingHandler.UpdateFromConfigMap)
-	} else if !apierrors.IsNotFound(err) {
-		logger.With(zap.Error(err)).Fatalf("Error reading ConfigMap %q", metrics.ConfigMapName())
-	}
+		// Watch the observability config map
+		if _, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(metrics.ConfigMapName(),
+			metav1.GetOptions{}); err == nil {
+			cmw.Watch(metrics.ConfigMapName(),
+				metrics.UpdateExporterFromConfigMap(component, logger),
+				profilingHandler.UpdateFromConfigMap)
+		} else if !apierrors.IsNotFound(err) {
+			logger.With(zap.Error(err)).Fatalf("Error reading ConfigMap %q", metrics.ConfigMapName())
+		}
 
-	if err := cmw.Start(ctx.Done()); err != nil {
-		logger.Fatalw("Failed to start configuration manager", zap.Error(err))
-	}
+		if err := cmw.Start(ctx.Done()); err != nil {
+			logger.Fatalw("Failed to start configuration manager", zap.Error(err))
+		}
 
-	// Start all of the informers and wait for them to sync.
-	logger.Info("Starting informers.")
-	if err := controller.StartInformers(ctx.Done(), informers...); err != nil {
-		logger.Fatalw("Failed to start informers", zap.Error(err))
-	}
+		// Start all of the informers and wait for them to sync.
+		logger.Info("Starting informers.")
+		if err := controller.StartInformers(ctx.Done(), informers...); err != nil {
+			logger.Fatalw("Failed to start informers", zap.Error(err))
+		}
 
-	// Start all of the controllers.
-	logger.Info("Starting controllers...")
-	go controller.StartAll(ctx.Done(), controllers...)
+		// Start all of the controllers.
+		logger.Info("Starting controllers...")
+		go controller.StartAll(ctx.Done(), controllers...)
+
+		profilingServer := profiling.NewServer(profilingHandler)
+
+		eg, egCtx := errgroup.WithContext(ctx)
+		eg.Go(profilingServer.ListenAndServe)
+
+		// If we have one or more admission controllers, then start the webhook
+		// and pass them in.
+		if len(webhooks) > 0 {
+			// Register webhook metrics
+			webhook.RegisterMetrics()
+
+			// possible bug? egCtx
```

Comment:
This looks like a new comment. Should @vagababov or @mattmoor comment on whether this ctx is correct?

Reply:
I'm pretty sure that this was intentional because the `ctx` here is being passed into work that is enqueued on the `errgroup`.  So it should key off of the SIGTERM not the first failure in the `errgroup`

Reply:
I believe it, it just looked a little fishy to me superficially, hence the note to myself.